site/app/controllers/admin/ReportController.php

```diff
@@ -776,8 +772,56 @@ public function displayGradebook() {
             new WebResponse(
                 ['admin', 'Report'],
                 'showFullGradebook',
-                $grade_file
+                $grade_file,
+                $grade_summaries_last_run
             )
         );
     }
+
+    /**
+     * Generate a custom filename for the downloaded CSV file
+     *
+     * @return string
+     */
+    private function generateCustomFilename() {
+        $course = $this->core->getConfig()->getCourse();
+        $timestamp = date("ymdHis");
+        return "{$course}_rainbow_grades_{$timestamp}.csv";
+    }
+
+
+    /**
+     * Download CSV file for Rainbow Grades
+     * @return void
+     */
+    #[Route("/courses/{_semester}/{_course}/reports/rainbow_grades_csv")]
+    public function downloadRainbowGradesCSVFile() {
+        // Check if the user has the necessary permissions (e.g., admin access)
+        if (!$this->core->getUser()->accessAdmin()) {
+            $this->core->getOutput()->showError("This account cannot access admin pages");
+        }
```

Comment:
I believe this isn't necessary since the entire class is annotated with `@AccessControl(role="INSTRUCTOR")`.